src/components/EmbeddedPlayground.tsx

```diff
@@ -298,29 +295,25 @@ function EmbeddedPlaygroundUI(props: { datastore: DataStore }) {
     ).editableContents!;
 
     // Invoke sharing.
-    const request = new ShareRequest();
-    request.setSchema(schema);
-    request.setRelationshipsYaml(relationshipsYaml);
-    request.setAssertionsYaml(assertionsYaml);
-    request.setValidationYaml(validationYaml);
-
-    service.share(
-      request,
-      {},
-      (err: grpcWeb.RpcError, response: ShareResponse) => {
-        if (err) {
+    try {
+    const { response } = await service.share({
+            schema,
+            relationshipsYaml,
+            assertionsYaml,
+            validationYaml,
+        });
+        const reference = response.shareReference;
+        window.open(`${window.location.origin}/s/${reference}`);
+    } catch (error: unknown) {
```

Comment:
This is what the new grpc-web invocations with the new generated code look like. It should be functionally identical.